staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go

```diff
@@ -91,22 +91,46 @@ func ValidateDeleteOptions(options *metav1.DeleteOptions) field.ErrorList {
 }
 
 func ValidateCreateOptions(options *metav1.CreateOptions) field.ErrorList {
-	return validateDryRun(field.NewPath("dryRun"), options.DryRun)
+	return append(
+		validateFieldManager(field.NewPath("fieldManager"), options.FieldManager),
+		validateDryRun(field.NewPath("dryRun"), options.DryRun)...,
+	)
 }
 
 func ValidateUpdateOptions(options *metav1.UpdateOptions) field.ErrorList {
-	return validateDryRun(field.NewPath("dryRun"), options.DryRun)
+	return append(
+		validateFieldManager(field.NewPath("fieldManager"), options.FieldManager),
+		validateDryRun(field.NewPath("dryRun"), options.DryRun)...,
+	)
 }
 
 func ValidatePatchOptions(options *metav1.PatchOptions, patchType types.PatchType) field.ErrorList {
 	allErrs := field.ErrorList{}
-	if patchType != types.ApplyPatchType && options.Force != nil {
-		allErrs = append(allErrs, field.Forbidden(field.NewPath("force"), "may not be specified for non-apply patch"))
+	if patchType != types.ApplyPatchType {
+		if options.Force != nil {
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("force"), "may not be specified for non-apply patch"))
+		}
+	} else {
+		if options.FieldManager == "" {
+			allErrs = append(allErrs, field.Required(field.NewPath("fieldManager"), "is required for apply patch"))
```

Comment:
Should it be required?

Reply:
Yes, but it's defaulted by kubectl. Controllers using it should use an explicit name.

Reply:
Maybe add a comment explaining why it's required.